README.md

```diff
@@ -2,302 +2,708 @@
+  field(:tags).type(:array)
 end
 ```
 
 Populate and use. Missing keys return defaults, if provided.
 
 ```ruby
-order_search = OrdersSearch.new(page: 2, q: 'foobar')
-order_search.params[:page] # => 2
-order_search.params[:per_page] # => 30
-order_search.params[:q] # => 'foobar'
-order_search.params[:status] # => nil
+form = schema.resolve(title: "A new blog post", tags: ["tech"])
+
+form.output # => {title: "A new blog post", tags: ["tech"], status: "draft"}
+form.errors # => {}
 ```
 
 Undeclared keys are ignored.
 
 ```ruby
-order_search = OrdersSearch.new(page: 2, foo: 'bar')
-order_params.params.has_key?(:foo) # => false
+form = schema.resolve(foobar: "BARFOO", title: "A new blog post", tags: ["tech"])
+
+form.output # => {title: "A new blog post", tags: ["tech"], status: "draft"}
 ```
 
+Validations are run and errors returned
+
+
 ```ruby
-order_search = OrderParams.new(status: 'checkout,closed')
-order_search.params[:status] #=> ['checkout', 'closed']
+form = schema.resolve({})
+form.errors # => {"$.title" => ["is required"]}
 ```
 
-### Search object pattern
-
-A class that declares allowed params and defaults, and builds a query.
+If options are defined, it validates that value is in options
 
 ```ruby
-class OrdersSearch
-  include Parametric::Params
-  param :q, 'Full text search query'
-  param :page, 'Page number', default: 1
-  param :per_page, 'Items per page', default: 30
-  param :status, 'Order status', options: ['checkout', 'pending', 'closed', 'shipped'], multiple: true
-  param :sort, 'Sort', options: ['updated_on-desc', 'updated_on-asc'], default: 'updated_on-desc'
+form = schema.resolve({title: "A new blog post", status: "foobar"})
+form.errors # => {"$.status" => ["expected one of draft, published but got foobar"]}
+```
+
+## Nested schemas
 
-  def results
-    query = Order.sort(params[:sort])
-    query = query.where(["code LIKE ? OR user_name LIKE ?", params[:q]]) if params[:q]
-    query = query.where(status: params[:status]) if params[:status].any?
-    query = query.paginate(page: params[:page], per_page: params[:per_page])
+A schema can have nested schemas, for example for defining complex forms.
+
+```ruby
+person_schema = Parametric::Schema.new do
+  field(:name).type(:string).required
+  field(:age).type(:integer)
+  field(:friends).type(:array).schema do
+    field(:name).type(:string).required
+    field(:email).validate(:email)
   end
 end
 ```
 
-### :match
+It works as expected
 
-Pass a regular expression to match parameter value. Non-matching values will be ignored or use default value, if available.
+```ruby
+results = person_schema.resolve(
+  name: "Joe",
+  age: "38",
+  friends: [
+    {name: "Jane", email: "jane@email.com"}
+  ]
+)
+
+results.output # => {name: "Joe", age: 38, friends: [{name: "Jane", email: "jane@email.com"}]}
+```
+
+Validation errors use [JSON path](http://goessner.net/articles/JsonPath/) expressions to describe errors in nested structures
```

Comment:
Well... The question is, what are people going to do with these paths? For an API this is fine, but if you wanted to use it to show validation errors on a rails form, I'm not sure how you'd do that nicely with either format (maybe it's not an issue).

In my experience json pointers are more common but that's basically because json schema uses them, I'm not sure what the rest of the world uses. I suspect there may be no right answer